openwisp_controller/config/api/views.py

```diff
@@ -125,6 +132,100 @@ class DeviceGroupDetailView(ProtectedAPIMixin, RetrieveUpdateDestroyAPIView):
     queryset = DeviceGroup.objects.select_related('organization').order_by('-created')
 
 
+def get_cached_devicegroup_args_rewrite(self, org_slugs, common_name):
+    url = reverse('config_api:devicegroup_x509_commonname', args=[common_name],)
+    url = f'{url}?org={org_slugs}'
+    return url
+
+
+# TODO: Think of a better identifier
+class DeviceGroupFromCommonName(ProtectedAPIMixin, RetrieveAPIView):
```

Comment:
the name is ok, or we could name it by common name instead of from. I think you can remove the TODO